test/config-luatest/replicaset.lua

```diff
@@ -50,6 +65,178 @@ end
+    end)
+
+    self:each(function(server)
+        server:wait_until_ready()
+    end)
+
+    self:each(function(server)
+        server:exec(function()
+            t.assert_equals(box.info.replicaset.name, 'replicaset-001')
+        end)
+    end)
+end
+
+-- Start the given instance.
+local function replicaset_start_instance(self, instance_name)
+    local server = self._server_map[instance_name]
+    assert(server ~= nil)
+    server:start()
+
+    server:exec(function()
+        t.assert_equals(box.info.replicaset.name, 'replicaset-001')
+    end)
+end
+
+local function replicaset_stop(self)
+    for _, server in ipairs(self._servers or {}) do
+        server:stop()
+    end
+end
+
+local function replicaset_drop(self)
+    for _, server in ipairs(self._servers or {}) do
+        server:drop()
+    end
+    self._servers = nil
+    self._server_map = nil
+end
+
+-- Sync the replicaset object with the new config.
+--
+-- It performs the following actions.
+--
+-- * Write the new config into the config file.
+-- * Update the internal list of instances.
+local function replicaset_sync(self, config)
+    local instance_names = instance_names_from_config(config)
+
+    treegen.write_script(self._dir, self._config_file_rel,
+                         yaml.encode(config))
+
+    for i, name in ipairs(instance_names) do
+        if self._server_map[name] == nil then
+            local server = server:new(fun.chain(self._server_opts, {
+                alias = name,
+            }):tomap())
+            table.insert(self._servers, i, server)
+            self._server_map[name] = server
+        end
+    end
+end
+
+-- Reload configuration on all the instances.
+local function replicaset_reload(self, config)
+    -- Rewrite the configuration file if a new config is provided.
+    if config ~= nil then
+        treegen.write_script(self._dir, self._config_file_rel,
+                             yaml.encode(config))
+    end
+
+    -- Reload config on all the instances.
+    self:each(function(server)
+        -- Assume that all the instances are started.
+        --
+        -- This requirement may be relaxed if needed, it is just
+        -- for simplicity.
+        assert(server.process ~= nil)
+
+        server:exec(function()
+            local config = require('config')
+
+            config:reload()
+        end)
+    end)
+end
+
+local methods = {
+    each = replicaset_each,
+    size = replicaset_size,
+    start = replicaset_start,
+    start_instance = replicaset_start_instance,
+    stop = replicaset_stop,
+    drop = replicaset_drop,
+    sync = replicaset_sync,
+    reload = replicaset_reload,
+}
+
+local replicaset_mt = {
+    __index = function(self, k)
+        if methods[k] ~= nil then
+            return methods[k]
+        end
+        if self._server_map[k] ~= nil then
+            return self._server_map[k]
+        end
+        return rawget(self, k)
+    end
+}
+
+local function new(g, config, server_opts)
```

Comment:
Comment: Maybe we can do smth about the fact, that logs are stored in another temporary directory, not inside the one, luatest creates? This makes logs checking difficult: find log of a test inside luatest created dirs, check the name of a dir, created by `treegen`, go there and find logs. `grep_log` also doesn't work without additional workarounds.

Reply:
It is on the `test.luatest_helpers.server` level. Can't remember why I initially don't integrate it better with `luatest.server` (I guess I met some problems on this road.) Anyway, I think that it is doable.

I would appreciate an issue with details about problems you have been observed.